lib/webenginesettings.cpp

```diff
@@ -109,11 +110,25 @@ void SailfishOS::WebEngineSettings::initialize()
     engineSettings->setPreference(QStringLiteral("intl.accept_languages"),
                                   QVariant::fromValue<QString>(langs));
 
+    Silica::Theme *silicaTheme = Silica::Theme::instance();
+
+    // Notify gecko when the ambience switches between light and dark
+    if (engineSettings->isInitialized()) {
+        engineSettings->notifyColorSchemeChanged();
+    } else {
+        connect(engineSettings, &QMozEngineSettings::initialized,
+                engineSettings, &SailfishOS::WebEngineSettings::notifyColorSchemeChanged);
+    }
+    connect(silicaTheme, &Silica::Theme::colorSchemeChanged,
+            engineSettings, &SailfishOS::WebEngineSettings::notifyColorSchemeChanged);
+
     // Guard preferences that should be written only once. If a preference needs to be
     // forcefully written upon each start that should happen before this.
     QString appConfig = QStandardPaths::writableLocation(QStandardPaths::AppDataLocation);
+    qDebug() << "PREFS: writing marker to: " << appConfig;
     QFile markerFile(QString("%1/__PREFS_WRITTEN__").arg(appConfig));
     if (markerFile.exists()) {
+        qDebug() << "PREFS: skipping";
```

Comment:
Remove these debugs before merging.

Reply:
Now removed. Thanks again for spotting these.